example/src/demos/Update.tsx

```diff
@@ -0,0 +1,166 @@
+import React, { useState, useRef, useEffect, useMemo, useLayoutEffect } from 'react'
+import { Canvas, FixedStage, Stage, Standard, useFrame, useThree, useUpdate } from '@react-three/fiber'
+import { a, useSpring } from '@react-spring/three'
+import { OrbitControls } from '@react-three/drei'
+import * as THREE from 'three'
+
+// ✅ Create a custom pipeline for advanced usecases.
+// ✅ Throw error if Update or Render stage is not included in a pipeline.
+// ✅ Add alpha to FixedStage and write interpolation example.
+// ✅ Add getStage method for interacting with Stages. Can get() and set() this way.
+// ✅ Add render with 'auto' | 'manual' flags to root. The priority flag is deprecated for the render flag.
+// ✅ The render function is executed inside the render stage.
+// ✅ Update useFrame to use mutation for subscriptions.
+// ✅ The useFrame loop is executed in the update stage.
+// ✅ Add a maxDelta in loops for tab safety.
+
+const colorA = new THREE.Color('#6246ea')
+const colorB = new THREE.Color('#e45858')
+
+// Quick and dirty HUD implementation.
+function useHud() {
+  const size = useThree((state) => state.size)
+
+  const hud = useMemo(() => {
+    const hudCanvas = document.createElement('canvas')
+    hudCanvas.width = size.width
+    hudCanvas.height = size.height
+    const hudBitmap = hudCanvas.getContext('2d')!
+    hudBitmap.font = '600 40px Inter'
+    hudBitmap.textAlign = 'center'
+    hudBitmap.fillStyle = 'rgba(245,10,10,0.75)'
+    hudBitmap.fillText('Canvas drawn HUD', size.width / 2, size.height / 2)
+    return hudCanvas
+  }, [size.height, size.width])
+
+  const camera = useMemo(
+    () => new THREE.OrthographicCamera(-size.width / 2, size.width / 2, size.height / 2, -size.height / 2, 0, 30),
+    [size.height, size.width],
+  )
+  const [scene] = useState(() => new THREE.Scene())
+  const texture = useMemo(() => new THREE.CanvasTexture(hud), [hud])
+  const material = useMemo(() => new THREE.MeshBasicMaterial({ map: texture, transparent: true }), [texture])
+  const mesh = useMemo(
+    () => new THREE.Mesh(new THREE.PlaneBufferGeometry(size.width, size.height), material),
+    [material, size.height, size.width],
+  )
+
+  useLayoutEffect(() => {
+    scene.add(mesh)
+  }, [scene, mesh])
+
+  return { scene, camera }
+}
```

Comment:
I'd simplify this demo a whole lot, even if there's "less" in it, to keep it easily parsable :)

Reply:
Makes sense. I was using this one to test features as I implemented them and just didn't clean it up.